src/usual/PLUsMgvStrat.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier:	AGPL-3.0
+pragma solidity ^0.8.14;
+
+import {ILiquidityProvider} from "mgv_src/strategies/interfaces/ILiquidityProvider.sol";
+import {IStratEvents} from "mgv_src/strategies/interfaces/IStratEvents.sol";
+import {UsualTokenInterface} from "mgv_src/usual/UsualTokenInterface.sol";
+import {MetaPLUsDAOToken} from "mgv_src/usual/MetaPLUsDAOToken.sol";
+import {MgvLib} from "mgv_src/MgvLib.sol";
+import {IERC20} from "mgv_src/IERC20.sol";
+import {IMangrove} from "mgv_src/IMangrove.sol";
+import {Direct} from "mgv_src/strategies/offer_maker/abstract/Direct.sol";
+import {MangroveOffer} from "mgv_src/strategies/MangroveOffer.sol";
+
+contract PLUsMgvStrat is Direct, IStratEvents {
+  UsualTokenInterface public immutable _pLUsDAOToken;
+  MetaPLUsDAOToken public immutable _metaPLUsDAOToken;
+  IERC20 public immutable _usUSD;
+  mapping(uint => address) offerIdToOwner; // keep track of who owns the offer
+  address public _usualDapp;
+  uint16 public _fee = 30; // Fee is in 10_000 = 100%
+  uint16 maxFee = 100;
+
+  constructor(
+    address admin,
+    IMangrove mgv,
+    UsualTokenInterface pLUsDAOToken,
+    MetaPLUsDAOToken metaPLUsDAOToken,
+    IERC20 usUSD
+  ) Direct(mgv, NO_ROUTER, 100_000) {
+    _pLUsDAOToken = pLUsDAOToken;
+    _metaPLUsDAOToken = metaPLUsDAOToken;
+    _pLUsDAOToken.approve(address(_metaPLUsDAOToken), type(uint).max);
+    _usUSD = usUSD;
+    if (admin != address(this)) {
+      setAdmin(admin);
+    }
+  }
+
+  modifier onlyDappOrAdmin() {
+    require(msg.sender == _usualDapp || msg.sender == admin(), "PLUsMgvStrat/onlyDappOrAdmin");
+    _;
+  }
+
+  function setFee(uint16 fee) public onlyAdmin {
+    require(fee <= maxFee, "PLUsMgvStrat/maxFee");
+    _fee = fee;
+    emit SetFee(fee);
+  }
+
+  function setUsualDapp(address usualDapp) public onlyAdmin {
+    _usualDapp = usualDapp;
+  }
+
+  function withdrawFees(address to) public onlyAdmin {
+    uint fee = _usUSD.balanceOf(address(this));
+    _usUSD.transfer(to, fee);
+    emit FeeWithdrawn(fee);
+  }
+
+  function __put__(uint amount, MgvLib.SingleOrder calldata order) internal virtual override returns (uint) {
+    uint fee = (amount * _fee) / 10_000;
+    _usUSD.transfer(offerIdToOwner[order.offerId], amount - fee);
+    emit FeePaid(fee);
+    return 0;
+  }
```

Comment:
I'd use internal functions `creditFee`, `debitFee` that log `CreditFee` and `DebitFee` and perform the required transfers, so that `withdrawFees` is simply an admin restricted wrapper of `debitFee` and `__put__` calls `creditFee`.

Reply:
Done